### DISCUSSION TITLE
TypeError: 'set' on proxy: trap returned falsish for property 'xxx'

### DISCUSSION THREAD

**Discussion by MijikHna** [vuejs/pinia#1486] [Help and Questions] [answered]
I am unable to access state properties from actions with this. I get always the error `TypeError: 'set' on proxy: trap returned falsish for property 'activeUser'`
here is the code of the store
```js
export const useAuthStore = defineStore("authStore", {
  state: () => ({
    activeUser: null as any,
    logoutTimer: null as any,
    isAuthenticated: false as boolean,
  }),
  getters: {
    activeUser: (state: any) => {
      return state.activeUser;
    },
    isAuthenticated: (state: any) => {
      return state.isAuthenticated;
    },
  },
  actions: {
    updateAuth(authenticated: boolean) {
      this.isAuthenticated = authenticated;
    },
    updateActiveUser(authUser: DecodesUserToken) {
      this.activeUser = authUser;
    },
  },
});
```
If I change `this.XXX` to `this.$state.XXX` in actions it works. I am using pinia verison *2.0.17*. Also in the components I am unable to access state attributes directly. I need to use `$state` before the actual state attribute.

**Comment by posva** [comment, marked as answer]
remove the activeUser and isAuthenticated getters, you don't need them in pinia. Normally this warns in the console as well.

**Reply by MijikHna** [reply]
Thanks, it works. But what If I decide to use getters where I will use some logic for example to filter state attributes. Was it because of the same key names in state and getters or do I always need to use `$state` if use getters?

**Reply by posva** [reply]
You have to name them differently, that’s all
